test/notification.cc

```diff
@@ -0,0 +1,56 @@
+#include "eventuals/notification.h"
+
+#include <future>
+
+#include "eventuals/do-all.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "test/promisify-for-test.h"
+
+namespace eventuals::test {
+namespace {
+
+TEST(NotificationTest, NotifyThenWait) {
+  Notification notification;
+
+  *notification.Notify();
+
+  *notification.Wait();
```

Comment:
Should we chain these two eventuals? (Why do we need two separate operator stars? My inference of eventuals style was to generally prefer to first build long chains, then execute the whole chain in one go)

Maybe add comments here if their separation is important.